snips_nlu_metrics/metrics.py

```diff
@@ -68,19 +72,66 @@ class must inherit from `Engine`
     global_confusion_matrix = None
     global_errors = []
     total_splits = len(splits)
-    for split_index, (train_dataset, test_utterances) in enumerate(splits):
+
+    def run_split(engine_class, split):
+        """
+            Fit and run engine on a split specified by train_dataset and
+            test_utterances
+        """
+        train_dataset, test_utterances = split
         engine = engine_class()
         engine.fit(train_dataset)
-        split_metrics, errors, confusion_matrix = compute_engine_metrics(
+        return compute_engine_metrics(
             engine, test_utterances, intent_list, include_slot_metrics,
             slot_matching_lambda)
-        global_metrics = aggregate_metrics(global_metrics, split_metrics,
+
+    def update_metrics(global_metrics,
+                       split_metrics,
+                       global_confusion_matrix,
+                       confusion_matrix,
+                       global_errors,
+                       errors):
+        """
+            Update global metrics with results on split
+        """
+        global_metrics = aggregate_metrics(global_metrics,
+                                           split_metrics,
                                            include_slot_metrics)
-        global_confusion_matrix = aggregate_matrices(global_confusion_matrix,
-                                                     confusion_matrix)
+        global_confusion_matrix = \
+            aggregate_matrices(global_confusion_matrix,
+                               confusion_matrix)
         global_errors += errors
-        if progression_handler is not None:
-            progression_handler(float(split_index + 1) / float(total_splits))
+        return global_metrics, global_confusion_matrix, global_errors
```

Comment:
If I'm correct, `run_split` and `update_metrics` functions can be extracted from `compute_cross_val_metrics` as they don't capture any local variable.
You can rename them with an `_` prefix to explicit the fact that they are not part of the public API.